tests/Kernel/TenderRenderTest.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Kernel;
+
+use Drupal\node\Entity\Node;
+use Drupal\Tests\oe_theme\PatternAssertions\ListItemAssert;
+use Drupal\Tests\oe_theme\PatternAssertions\FieldListAssert;
+use Drupal\Tests\oe_theme\PatternAssertions\PatternAssertState;
+use Drupal\Tests\user\Traits\UserCreationTrait;
+use Drupal\user\Entity\User;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests the event rendering.
+ */
+class TenderRenderTest extends ContentRenderTestBase {
+
+  use UserCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'field_group',
+    'composite_reference',
+    'oe_time_caching',
+    'oe_content_reference_code_field',
+    'oe_content_departments_field',
+    'oe_content_tender',
+    'oe_theme_content_tender',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installConfig([
+      'composite_reference',
+      'oe_content_departments_field',
+      'oe_content_reference_code_field',
+      'oe_content_tender',
+      'oe_theme_content_tender',
+    ]);
+
+    module_load_include('install', 'oe_theme_content_tender');
+    oe_theme_content_tender_install();
+
+    module_load_include('install', 'oe_content');
+    oe_content_install();
+
+    // Set current user to UID 1, so that by default we can access everything.
+    $account = User::load(1);
+    $this->setCurrentUser($account);
+  }
+
+  /**
+   * Test an event being rendered as a teaser.
+   */
+  public function testTenderTeaser(): void {
+    // Create a Call for tender node.
+    /** @var \Drupal\node\Entity\Node $node */
+    $values = [
+      'type' => 'oe_tender',
+      'title' => 'Test Call for tenders node',
+      'oe_publication_date' => [
+        'value' => '2020-04-15',
+      ],
+      'oe_tender_opening_date' => [
+        'value' => '2020-04-30',
+      ],
+      'oe_tender_deadline' => [
+        'value' => date('Y') + 1 . '-06-10T23:30:00',
```

Comment:
We need to use time mocking here, as we did on events and other content type. `Check \Drupal\Tests\oe_theme\Kernel\EventRenderTest::testEventTeaser()`, we need to use that, and not fiddle with years in the tests itself.

This how it would look like:

![image](https://user-images.githubusercontent.com/153362/93756609-a3b7f700-fc05-11ea-86d9-caf9338ae662.png)

Reply:
Updated